### PR TITLE
asciidoctor: use ruby30 on macOS 10.7 and 10.8

### PR DESCRIPTION
**Pull request by aeiouaeiouaeiouaeiouaeiouaeiou**
Closes: https://trac.macports.org/ticket/66409

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.8.5 12F2560 x86_64
Xcode 5.1.1 5B1008

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
@aeiouaeiouaeiouaeiouaeiouaeiou This gonna break `asciidoctor` for all PPC systems though. (But if `ruby31` won’t be fixed any soon for 10.8, I can make patches for `ruby30` to make it build on PPC.)

**Comment by barracuda156** [comment]
> if ruby31 cannot be fixed on these old systems then just restrict this change to these specific systems.

This is more sensible indeed.

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
@reneeotten please review

File: textproc/asciidoctor/Portfile [hunk 1]
```diff
@@ -5,7 +5,13 @@ PortGroup           ruby 1.0
 
 ruby.setup          asciidoctor 2.0.18 gem {} rubygems ruby31
 name                asciidoctor
-revision            1
+revision            2
```
Comment (reneeotten) [inline textproc/asciidoctor/Portfile:8]: no need to increase the "revision" as you are fixing a build issue. Everything that did build doesn't change..

**Review by reneeotten — CHANGES_REQUESTED** [review]
sorry, I forgot about this... one last change and then this is good to go.

**Comment by fhgwright** [comment]
> @aeiouaeiouaeiouaeiouaeiouaeiou This gonna break `asciidoctor` for all PPC systems though. (But if `ruby31` won’t be fixed any soon for 10.8, I can make patches for `ruby30` to make it build on PPC.)

Well, ruby30 will be fine once #17386 is merged. :-)

Note that, when I originally tried it, ruby31 built on 10.4 ppc but not 10.5 ppc.  It now builds on 10.5 ppc (with no change to ruby31 itself), perhaps because I wrestled with getting gcc7 updated (a build that takes almost 21 hours on the G4 mini).

Also, here I see ruby31 building on 10.7 but not 10.8 or 10.9, so in general it isn't very consistent.

It shouldn't be one port's responsibility to keep track of which systems its dependencies build on, unless the dependencies have very clearly stated compatibility requirements.  It's better just to pick a version that works across the board and stick with it until a better version that works across the board comes along.

Asciidoctor itself doesn't particularly care what ruby version it runs with, as long as it's something that works.  The ruby version requirement stated on its website is ">= 0".  However, the RaspberryPi documentation site indicates that asciidoctor should be run with ruby 2.7 because it doesn't work properly with 3.0.  Though that recommendation probably predates ruby 3.1, so it's not clear whether its issue is with == 3.0 or >= 3.0.  It's also based on getting everything from homebrew, which may or may not matter.
Sticking with 2.7 might be safer, and ruby27 will also be fixed with #17385.  But I haven't attempted to figure out what the RaspberryPi folks are complaining about.

Avoiding a revbump is inappropriate when build failures aren't consistent.  And it's pretty silly to be stingy with revbumps on a port that literally takes five seconds to build on a decent machine.  Even on the G4 mini it only takes 32s.

**Comment by barracuda156** [comment]
@fhgwright As I recall, 2.7 does not build with PPC either: patches are needed to make it work. There was a ticket on Trac with discussion of that.

3.11 was tested with gcc11 and gcc12. If it fails to build with <7.5.0, well, that is not a fault of PPC, I guess. Also, we should move PPC systems to gcc11/gcc12, hopefully soon. There is little to no need to carry on with compilers which bugs are not fixed anymore.

**Comment by reneeotten** [comment]
> Avoiding a revbump is inappropriate when build failures aren't consistent. And it's pretty silly to be stingy with revbumps on a port that literally takes five seconds to build on a decent machine. Even on the G4 mini it only takes 32s.

well... a build either fails or doesn't - there shouldn't be anything inconsistent about that. And when fixing a build failure the `revision` should not be increased. Whether or not a port builds quickly or takes forever doesn't change that.

**Comment by fhgwright** [comment]
> > Avoiding a revbump is inappropriate when build failures aren't consistent. And it's pretty silly to be stingy with revbumps on a port that literally takes five seconds to build on a decent machine. Even on the G4 mini it only takes 32s.
> 
> well... a build either fails or doesn't - there shouldn't be anything inconsistent about that. And when fixing a build failure the `revision` should not be increased. Whether or not a port builds quickly or takes forever doesn't change that.

But in this case it's not a direct build failure in the port, but rather a build failure in a dependency, which doesn't seem to be consistent for a given OS version.  Thus, without a revbump, the same revision of the same version of the same port on the same OS version may use one version of Ruby on some machines, and a different version on others, which is inconsistent.

**Comment by fhgwright** [comment]
I investigated the alleged RaspberryPi issue.  This was based on:

`**NOTE:** Homebrew defaults to Ruby 3.0 which is incompatible with Asciidoctor.
`

When following the relevant subset of the instructions, I noted that `bundle install` brings in `asciidoctor-2.0.15` as a gem.  It wasn't immediately obvious whether that version or the MacPorts version would get used.

The build procedure isn't sufficiently informative, and it doesn't have a useful verbose mode, but I was able to use tracing to determine that:

1. The version of Ruby specified by `port select` is the one that gets used.
2. The "bundle" version of `asciidoctor` is the one that gets used.
3. It's `asciidoctor-2.0.15` whether `ruby27` or `ruby30` gets used.
4. It works fine in either case.

Thus, the `rubyXX` selection for the `asciidoctor` port doesn't even affect the RaspberryPi documentation build, and even if it did, it would be fine either way.  So `ruby30` is the obvious choice, once `ruby30` is fixed.

I'd already planned to submit a PR to do that once #17386 is merged, including the usual documented wide testing.

**Comment by fhgwright** [comment]
I've now fully tested `asciidoctor` with `ruby30`.  I pushed it to my fork, but didn't submit a PR since it conflicts with this one, and it shouldn't be merged until after #17386.  See https://github.com/fhgwright/macports-ports/tree/asciidoctor.

**Comment by fhgwright** [comment]
@aeiouaeiouaeiouaeiouaeiouaeiou 
If you close this PR, I'll open one with my version, and mark it as a draft until #17386 is merged.

**Comment by fhgwright** [comment]
Opened #17519 as the replacement.

**Pull request closed without merging**